File: testing/logic-error/logic_error.vb
' Raspberry Pi Foundation
' Usage licensed under CC BY-SA 4
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program

Module Program

    ' The Main subroutine is the default entry point for a VB program
    Sub Main()
        Dim amount As Double = 130.4
        Dim result As Double = AddVat(amount)
        Console.WriteLine(result)
    End Sub

    Function AddVat(ByVal amount As Double) As Double
        ' Demonstrates a logic error
        Dim RATE As Integer = 20
        Dim vat As Double = amount * RATE
        Dim withVat As Double = amount + vat
        Return withVat
    End Function

End Module

File: testing/logic-error/java/logic_error.java
/*
Raspberry Pi Foundation
Developed as part of Ada Computer Science

Usage licensed under CC BY-NC-SA 4.0

Note: This file is designed to be copied out and compiled on your machine.
In order for the program to compile properly in some IDEs, you need to ensure that the
class filename is the same as the class name with the main method in it.

To run this file you need to:
1. Copy the contents
2. Create a Java project in the IDE of your choice (Eclipse, for example)
3. Create a new Java Class that uses the same name as the class containing the main method in this program
4. Paste the contents into the new Java Class
5. Save and run the program
*/

class LogicErrors
{
    // The main method is the entry point for all Java programs
    public static void main(String[] args) {
        double amount = 130.40;
        double result = addVat(amount);
        System.out.println(result);
    }


    // Calculates and applies VAT at standard rate
    public static double addVat(double amount) {
        // Demonstrates a logic error
        int RATE = 20;
        double vat = amount * RATE;
        double withVat = amount + vat;
        return withVat;
    }


}